### PR TITLE
adding support for sample type in the report encoder

### PR DESCRIPTION
**Pull request by russelldeguzman**
To support additional sensors (i.e. BM soft) when we need to know what type of sample to decode/bitpack on the Spotter. 
Therefore, I added a string at the top of each sample array to define what kind of samples are contained in the array. 

On the Spotter side - I'll generalize a report builder decode library and account for this extra string member in the sample array.

### CODE REVIEW

File: src/apps/bridge/reportBuilder.cpp [hunk 1]
```diff
@@ -323,70 +349,90 @@ void reportBuilderAddToQueue(uint64_t node_id, uint8_t sensor_type, void *sensor
...
-        BRIDGE_LOG_PRINT("Failed to close sample in addSamplesToReport\n");
-        break;
-      }
-      rval = true;
+  switch (sensor_type) {
+  case AANDERAA_SENSOR_TYPE: {
+    aanderaa_aggregations_t aanderaa_sample =
+        (static_cast<aanderaa_aggregations_t *>(sensor_data))[sample_index];
+    if (sensor_report_encoder_open_sample(context, AANDERAA_NUM_SAMPLE_MEMBERS,
+                                          "aandera_current_v0") != CborNoError) {
```
Comment (russelldeguzman) [inline src/apps/bridge/reportBuilder.cpp]: This is the only "real" change in the file. 
The rest is `clang-format`

File: src/apps/bridge/cbor_sensor_report_encoder.cpp [hunk 2]
```diff
@@ -87,14 +87,25 @@ CborError sensor_report_encoder_close_sensor(sensor_report_encoder_context_t &co
 }
 
 /*!
- * @brief Open a sample array for encoding.
+ * @brief Open a sample array for encoding. The first member in the sampe array is the sample type.
```
Comment (victorsowa12) [inline src/apps/bridge/cbor_sensor_report_encoder.cpp]: nit: sampe -> sample

**Review by victorsowa12 — APPROVED** [review]
Small nit but looks good!

File: src/apps/bridge/reportBuilder.cpp (hunk 1 above)
Comment (towynlin) [reply src/apps/bridge/reportBuilder.cpp]: nit: aandera -> aanderaa

**Review by towynlin — APPROVED** [review]
similarly, looks good — found one more typo

**Pull request merged**